Print right hex value for MIN_INT

catenacyber opened the pull request:
Found using oss-fuzz

You cannot compute `- INT_MIN` as it is equal to `INT_MAX+1`

aquynh commented:
merged, thanks!

could you please double check with the master branch? this bug was fixed there, but i think this is a better fix.

catenacyber commented:
> could you please double check with the master branch? this bug was fixed there, but i think this is a better fix.

see #1160
I deleted the fix and added this new fix in one commit
